cmd/kube-proxy/app/server.go

```diff
@@ -393,17 +393,22 @@ type ProxyServer struct {
 // createClients creates a kube client and an event client from the given config and masterOverride.
 // TODO remove masterOverride when CLI flags are removed.
 func createClients(config componentconfig.ClientConnectionConfiguration, masterOverride string) (clientset.Interface, v1core.EventsGetter, error) {
-	if len(config.KubeConfigFile) == 0 && len(masterOverride) == 0 {
-		glog.Warningf("Neither --kubeconfig nor --master was specified. Using default API client. This might not work.")
-	}
+	var kubeConfig *rest.Config
+	var err error
 
-	// This creates a client, first loading any specified kubeconfig
-	// file, and then overriding the Master flag, if non-empty.
-	kubeConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		&clientcmd.ClientConfigLoadingRules{ExplicitPath: config.KubeConfigFile},
-		&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: masterOverride}}).ClientConfig()
-	if err != nil {
-		return nil, nil, err
+	// When only master URL is set, use service account but override the host.
+	if len(config.KubeConfigFile) == 0 && len(masterOverride) != 0 {
+		glog.Info("Only --master was specified. Using service account with overrided host.")
```

Comment:
If we want to keep backwards-compat, we can't do this.
`--master` only currently means "I can run anywhere, and I will connect insecurely (via 8080 port) to the API server"
I guess we need to add yet an other `--use-service-account-credentials` or similar flag :/

Reply:
> I guess we need to add yet an other --use-service-account-credentials or similar flag :/

Thanks, I think you are right...

Reply:
Added a flag `--service-account-master-url`. PTAL, thanks!